Project Files/DLLSources/CvCity.cpp

```diff
@@ -11044,11 +11032,30 @@ void CvCity::doLbD()
 	int train_percent = GC.getGameSpeedInfo(GC.getGameINLINE().getGameSpeedType()).getTrainPercent();
 
 	//moddifying values with GameSpeed
-	chance_increase_expert = chance_increase_expert / train_percent / 100;
+	chance_increase_expert = chance_increase_expert * 100 / train_percent;
 	pre_rounds_expert = pre_rounds_expert * train_percent / 100;
 
-	chance_increase_free = chance_increase_free / train_percent / 100;
+	chance_increase_free = chance_increase_free * 100 /  train_percent ;
+	if(chance_increase_free <= 0) {
+		chance_increase_free = 1;
+	}
+	if (chance_increase_expert <= 0) {
+		chance_increase_expert = 1;
+	}
 	pre_rounds_free = pre_rounds_free * train_percent / 100;
+	for (int iTrait = 0; iTrait < GC.getNumTraitInfos(); ++iTrait)
+	{
+		TraitTypes eTrait = (TraitTypes) iTrait;
+		if (eTrait != NO_TRAIT)
+		{
+			if (hasTrait(eTrait))
+			{
+				pre_rounds_expert *= 100;
+				pre_rounds_expert /= GC.getTraitInfo(eTrait).getLearningByDoingModifier() + 100;

```

Comment:
The math is done differently from math elsewhere. Usually it's (100 - modifier)/100. Here it's 100/(modifier + 100). Why this difference?

Another note is to add int iModifer = GC.getTraitInfo(eTrait).getLearningByDoingModifier() and then only do the */ if iModifer != 0. The approach here will result in a lot of *100 /100 operations.

Reply:
Why the value is 100/(modifier+100) : Because the modifier is an increase in rate. If I say learning is 100% faster, I mean that it will take half the time to learn it. If you use (100-modifier)/100, you will end with zero time learning...